jinja2/utils.py

```diff
@@ -284,6 +284,25 @@ def generate_lorem_ipsum(n=5, html=True, min=20, max=100):
     return Markup(u'\n'.join(u'<p>%s</p>' % escape(x) for x in result))
 
 
+def native_concat(invals):
+    '''
+    Jinja's native concat is just a list join. The list comes from result of
+    templating all the nodes. If the return type is expected to be
+    native, the list is artificial and we should return just the first
+    element.
+    '''
+    invals = [x for x in invals]
+    if len(invals) == 1:
+        # break out the single value
+        invals = invals[0]
+    elif len(invals) > 1:
+        # cast to unicode and join
+        try:
+            invals = u''.join([text_type(x, 'utf8') for x in invals])
```

Comment:
Never mind, I messed up the review for this one. _Should_ use `text_type`, which is what `to_string` is in `runtime`. But _don't_ use `utf8`, since that's only relevant if the object is `bytes`. Should remove the `except Exception` block.